blockreassurance.php

```diff
@@ -102,11 +95,19 @@ public function __construct()
 
         $this->bootstrap = true;
         parent::__construct();
+        if ($this->context->link == null) {
+            $protocol_link = (Tools::usingSecureMode() && Configuration::get('PS_SSL_ENABLED')) ? 'https://' : 'http://';
```

Comment:
Here, `$protocol_link` and `$protocol_content` have the same content.
Why not use `Tools::getCurrentUrlProtocolPrefix()`? And I think you only need to check if the server is properly configured, no need to check for PS_SSL_ENABLED.

Reply:
:ok_hand: